### PR TITLE
Parole population by district - map viz

### LINKED ISSUE #10: [Parole page] Proportional symbol map for population
Inputs:
- GeoJSON shape file of ND counties (find one published)
- Geo-coordinates of parole district office locations (from backend)
- Aggregate data of parole population by district (from backend)

Output:
- map of ND
- Population by district as proportional circle, centered on office location

Behavior:
- Currently selected district (from external control) should be highlighted
- Clicking on a district's proportional symbol selects that district (via external control)

### PR DESCRIPTION
**Pull request by macfarlandian**
## Description of the change

Creates the first chart component and a first pass at a standardized way that all chart components will receive props (which may contain data they need to render charts, control states, and control state update functions). I didn't fuss over this too much because it will almost certainly change; for now it is hopefully relatively lightweight and modular. (The general idea is data sources are specified in the page configuration object, and all viz components will receive all of the control props, which they can use or ignore as needed.)

Screen Shot 2020-07-07 at 11 44 10 AM

Interaction polish will follow later but for now there are basic hover states on the district markers, and clicking them should update the selected district. (I used the same underlying mapping library that we use in pulse-dashboard, though I believe it's a newer version because the API has changed slightly. I also copied over the ND topography file from pulse-dashboard.)

Untracked local data files are used for testing at this point in lieu of a backend; you will need them to test this locally if you're planning to do that, contact me and I shall provide.

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Configuration change (adjusts configuration to achieve some end related to functionality, development, performance, or security)

## Related issues

Closes #10 

### Development

These boxes should be checked by the submitter prior to merging:

- [x] Manual testing against realistic data has been performed locally

### Code review

These boxes should be checked by reviewers prior to merging:

- [x] This pull request has a descriptive title and information useful to a reviewer
- [x] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [x] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

File: src/page-parole/PageParole.js [hunk 1]
```diff
@@ -1,5 +1,11 @@
 import React from "react";
 import DetailPage from "../detail-page";
+// once the backend is in place, stop using these untracked test files
```
Comment (jessex) [inline src/page-parole/PageParole.js:3]: @macfarlandian, when you share your care package with @sprad, make sure to send him these files, as well. Thanks!

File: src/viz-parole-population/VizParolePopulation.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  const districtTotals = populationDemographics
+    .filter(
+      (record) =>
+        record.race_or_ethnicity === TOTAL_KEY &&
+        record.gender === TOTAL_KEY &&
+        record.age_bucket === TOTAL_KEY
+    )
+    .map((record) => {
+      const districtData = districtOffices.find(
+        (office) => +office.district === +record.district
```
Comment (jessex) [inline src/viz-parole-population/VizParolePopulation.js:35]: Not sure I'm familiar with this syntax. What does the prepended plus sign do?

File: src/viz-parole-population/VizParolePopulation.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        return {
+          district: `${record.district}`,
+          lat: districtData.lat,
+          long: districtData.long,
+          value: +record.total_supervision_count,
+        };
+      }
+      return null;
+    })
+    .filter((record) => record);
```
Comment (jessex) [inline src/viz-parole-population/VizParolePopulation.js:47]: Why is this filter statement needed?

File: src/viz-parole-population/VizParolePopulation.js (hunk 2 above)
Comment (macfarlandian) [reply src/viz-parole-population/VizParolePopulation.js:35]: cheater method for casting them both to numbers ... looking at it again though i'm not sure it's really warranted, I should probably try to standardize them as strings instead

File: src/viz-parole-population/VizParolePopulation.js (hunk 3 above)
Comment (macfarlandian) [reply src/viz-parole-population/VizParolePopulation.js:47]: this will filter out any `null`s that may have resulted the previous `map` result

**Pull request merged**